File: DUNG-2010/luyende3/kitu.py
s = input()
a = int(input())
n = len(s)
stack = []
for digit in s:
 
    while a > 0 and stack and stack[-1] > digit:
        stack.pop()
        a -= 1
    stack.append(digit)


if a > 0:
    stack = stack[:-a]

res = "".join(stack)

res = res.lstrip('0')

print(res if res else "0")

File: KHOI-2010/KITU.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
    freopen("KITU.INP", "r", stdin);
    freopen("KITU.OUT", "w", stdout);
    string s;
    int k;
    cin >> s;
    cin >> k;
    stack<char> st;
    for (char c : s) {
        while (!st.empty() && k > 0 && st.top() > c) {
            st.pop();
            k--;
        }
        st.push(c);
    }
    while (k > 0 && !st.empty()) {
        st.pop();
        k--;
    }
    string res = "";
    while (!st.empty()) {
        res += st.top();
        st.pop();
    }
    reverse(res.begin(), res.end());
    int i = 0;
    while (i < res.size() && res[i] == '0') i++;
    res = res.substr(i);
    if (res == "") res = "0";
    cout << res;
    return 0;
}